src/pages/home/ReportScreen.js

```diff
@@ -62,7 +63,11 @@ class ReportScreen extends React.Component {
      */
     getReportID() {
         const params = this.props.route.params;
-        return Number.parseInt(params.reportID, 10);
+        const id = Number.parseInt(params.reportID, 10);
+        if (_.isNaN(id)) {
+            handleInaccessibleReport();
```

Comment:
After looking at this and testing I'm not sure this is the best place to put this check. This function will now redirect to the concierge chat whenever we call `getReportID()` to check the reportID and the value is not a number, so checks like [this](https://github.com/Expensify/App/pull/4387/files#diff-dd4bfa50713397aca8cb40145317936ab0affe19abc2a9c24d6c8849ed75dc9bR79) can now redirect when the intention is to get the value.

I think a better way to handle this would be to move the redirect to the scope that calls `getReportID`. So we'd call `getReportID()` and check the return value, and if the return value is NaN we'd do the redirect. Does that make sense?

Reply:
@Jag96 
Agreed, a function should behave however it's define.

> we'd call getReportID() and check the return value, and if the return value is NaN we'd do the redirect.

Then, we'll have to add the checks in all the locations where we want to prevent this behaviour. (Code will not be **DRY**)

* I think that we can follow the same convention used here 
   https://github.com/Expensify/App/blob/d15beee9b01c782b60f7cb47de795bc95c02b864/src/libs/actions/Report.js#L312-L321

* The function will look like this
```js
      /**
     * Get the currently viewed report ID as number
     * If the shouldRedirectIfNan flag is set, we redirect to the Concierge chat
     * when the ID is not a number
     * @param {Boolean} shouldRedirectIfNan
     * @returns {Number}
     */
    getReportID(shouldRedirectIfNan = false) {
        const params = this.props.route.params;
        const id = Number.parseInt(params.reportID, 10);
        if (shouldRedirectIfNan && _.isNaN(id)) {
            handleInaccessibleReport();
        }
        return id;
    }
```


* Passing the check here seems good to me.
  https://github.com/Expensify/App/blob/d15beee9b01c782b60f7cb47de795bc95c02b864/src/pages/home/ReportScreen.js#L90-L93

Reply:
> Then, we'll have to add the checks in all the locations where we want to prevent this behaviour. (Code will not be DRY)

I agree with this approach if there will be multiple places where we'll pass a value for `shouldRedirectIfNan` into getReportID, however, if we're only going to pass that value in one place I think it's better to just have the `_.isNaN` check inside `storeCurrentlyViewedReport` after we get the reportID. Then, if we need to add more usages of this later we can DRY it by updating the function the way you suggested. Thoughts?

Reply:
> if we need to add more usages of this later we can DRY it by updating the function the way you suggested.

Yeah, sounds good.